### PR TITLE
IRCv3 message tags

### LINKED ISSUE #34: Add support for IRCv3 message tags
See http://ircv3.net/specs/core/message-tags-3.2.html

### PR DESCRIPTION
**Pull request by belak**
This should close #34

### CODE REVIEW

**Comment by belak** [comment]
This implements support for http://ircv3.net/specs/core/message-tags-3.2.html

File: parser.go [hunk 1]
```diff
@@ -5,6 +5,123 @@ import (
 	"strings"
 )
 
+var tagDecodeSlashMap = map[rune]rune{
+	':':  ';',
+	's':  ' ',
+	'\\': '\\',
```
Comment (jsvana) [inline parser.go]: I don't follow this, why switch from the same character to itself?

File: parser.go [hunk 2]
```diff
@@ -5,6 +5,123 @@ import (
...
+	'\n': "\\n",
+}
+
+// TagValue represents the value of a tag.
+type TagValue string
+
+// ParseTagValue parses a TagValue from the connection. If you need to
+// set a TagValue, you probably want to just set the string itself, so
+// it will be encoded properly.
+func ParseTagValue(v string) TagValue {
```
Comment (jsvana) [inline parser.go]: there's no way to automatically unescape values in golang? this seems like it's been written before

File: parser.go [hunk 3]
```diff
@@ -5,6 +5,123 @@ import (
...
+	ret, ok := t[key]
+	return string(ret), ok
+}
+
+// String ensures this is stringable
+func (t Tags) String() string {
+	buf := &bytes.Buffer{}
+
+	for k, v := range t {
+		buf.WriteByte(';')
```
Comment (jsvana) [inline parser.go]: why `Byte` here and `Rune` above?

Comment (belak) [inline parser.go]: Byte is slightly faster as a rune can contain multiple bytes.

File: parser.go (hunk 2 above)
Comment (belak) [inline parser.go]: I haven't been able to find one, but I'd be happy to change it if we find one later.

File: parser.go (hunk 1 above)
Comment (belak) [inline parser.go]: This is so '\\' will decode to '\' (tagDecodeSlashMap is the character that comes *after* a slash.

**Pull request merged**